### ISSUE TITLE
BeagleTreeLikelihood with multiple instances on multiple threads returns -Inf

### ISSUE THREAD

**Issue by armanbilge** [armanbilge/BEAST3#21]
When using BeagleTreeLikelihood with >1 instances in parallel, often the likelihood initializes to -Inf. This behaviour is non-deterministic, even with a fixed seed. Even the specific instances that return -Inf change between runs. When they do not return -Inf, they consistently return the correct likelihood. This issue emerges only when there are also >1 threads, suggesting a multithreading problem.

```
  CompoundLikelihood(compoundModel)=(
    BeagleTreeLikelihood(treeLikelihood[mcmc.treeLikelihood_0])=-Inf, 
    BeagleTreeLikelihood(treeLikelihood[mcmc.treeLikelihood_1])=-107.546
    Total = -Infinity
  )
```

**Comment by armanbilge** [comment]
I believe that this issue is a duplicate of https://code.google.com/p/beast-mcmc/issues/detail?id=750 .
